src/Symfony/Component/DependencyInjection/Argument/AbstractableArgument.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+/**
+ * Represents an abstract parameter, which have to be set by a compiler pass or a DI extension.
+ */
+final class AbstractableArgument
+{
+    private $serviceId;
+    private $argKey;
+    private $text;
+
+    public function __construct(string $serviceId, string $argKey, string $text = '')
+    {
+        $this->serviceId = $serviceId;
+        $this->argKey = $argKey;
+        $this->text = $text;
+    }
+
+    public function getServiceId(): string
+    {
+        return $this->serviceId;
+    }
+
+    public function getArgumentKey(): string
+    {
+        return $this->argKey;
+    }
+
+    public function getText(): string
+    {
+        return $this->text;
+    }
+
+    public function __toString(): string
```

Comment:
I think we don't need this method - instead, messages should be computed inline when needed, using `getText()`, especially because part of them is already computed inline.

Reply:
`getText()` can return an empty string, and this will have to be checked for each message computing to remove a colon in front of text

Reply:
yes, and that's fine to me :)

Reply:
done